applications/Preprocessor/centaur.cpp

```diff
@@ -257,313 +203,160 @@ CentaurReader::CentaurReader(std::string filename) {
     //  - 1 data line with N zone family names
     //
     // Note 2D: Version prior <= 4.99 allow not having zones it seems
-    readZoneInfo(elementCounts);
+    readZoneInfo();
     logger(INFO, "Finished scanning through hyb-file");
 
     // More internal information is possible, for example polyhedrons.
 }
 
-std::function<void(int&)> difference_generator(
-    const std::vector<std::size_t>& input) {
-    std::size_t i = 1;
-    return [=](int& result) mutable {
-        result = static_cast<long long>(input[i + 1]) -
-                 static_cast<long long>(input[i]);
-        i++;
-    };
-}
+void CentaurReader::readHeader() {
+    // We do not apriori know the size of the header
+    std::uint32_t headerSize = centaurFile.peekRecordSize();
+    std::vector<char> rawInput(headerSize);
+
+    centaurFile.readRawRecord(headerSize, rawInput.data());
+    std::uint32_t offset = 0;
+    version = *reinterpret_cast<typeof(version)*>(rawInput.data());
```

Comment:
I am not a 100% sure about this but I think we should use `memcpy` not `reinterpret_cast` https://www.reddit.com/r/cpp/comments/gb2oz1/convert_bytes_to_object_memcpy_vs_reinterpret/

Reply:
Reading through the documentation on reinterpret_cast here (https://en.cppreference.com/w/cpp/language/reinterpret_cast) it indeed looks that way. As it explicitly notes that you are allowed to cast to `char*` and as well as

> When it is needed to interpret the bytes of an object as a value of a different type, std::memcpy or std::bit_cast (since C++20)can be used

So indeed I have to use memcpy here.

Thanks for catching that.

Reply:
I recently saw it in a talk, but this is really obscure.